### PR TITLE
Add GA Logging to Portal

### PR DESCRIPTION
**Pull request by tomhschmidt**
## Description

Added Google Analytics event logging for key user activity in the portal.

## Motivation and Context

Allows us to better understand what actions users are taking when they go to portal beyond existing traffic numbers. Can also be used to find errors and bugs in the future.

## How Has This Been Tested?

1. Go to allowances page.
2. Set allowance for WETH and ZRX
3. Verify that allowance events for WETH and ZRX have been logged

https://analytics.google.com/analytics/web/?authuser=1#report/content-event-overview/a98720122w145076439p149780311/%3F_u.date00%3D20180205%26_u.date01%3D20180205%26overview-dimensionSummary.selectedGroup%3Dvisitors%26overview-dimensionSummary.selectedDimension%3Danalytics.eventLabel/

1. Go to orders page
2. Sign order
3. Verify that sign order events have been logged

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

* [ ] Change requires a change to the documentation.
* [ ] Added tests to cover my changes.

### CODE REVIEW

File: packages/website/ts/components/inputs/allowance_toggle.tsx [hunk 1]
```diff
@@ -74,7 +75,24 @@ export class AllowanceToggle extends React.Component<AllowanceToggleProps, Allow
...
+                },
+                () => {
+                    ReactGA.event({
+                        category: 'Portal',
+                        action: 'Set Allowance Failure',
+                        label: this.props.token.symbol,
+                        value: newAllowanceAmountInBaseUnits.toNumber(),
+                    });
+                },
+            );
```
Comment (fabioberger) [inline packages/website/ts/components/inputs/allowance_toggle.tsx]: No need to use `.then` here, we can just call these two calls after we've finished awaiting the `setProxyAllowanceAsync` call. Any specific reason you used promises here?

File: packages/website/ts/components/inputs/allowance_toggle.tsx [hunk 2]
```diff
@@ -74,9 +75,21 @@ export class AllowanceToggle extends React.Component<AllowanceToggleProps, Allow
             newAllowanceAmountInBaseUnits = DEFAULT_ALLOWANCE_AMOUNT_IN_BASE_UNITS;
         }
         try {
-            await this.props.blockchain.setProxyAllowanceAsync(this.props.token, newAllowanceAmountInBaseUnits);
+            await this.props.blockchain.setProxyAllowanceAsync(this.props.token, newAllowanceAmountInBaseUnits)
```
Comment (fabioberger) [inline packages/website/ts/components/inputs/allowance_toggle.tsx]: Add missing semi-colon. Do you have a prettier extension installed in your text editor?

Comment (tomhschmidt) [reply packages/website/ts/components/inputs/allowance_toggle.tsx]: Oops. I do not -- will fix and install

**Comment by tomhschmidt** [comment]
@fabioberger Yeah, I can add order fill and order cancel. I'm trying to keep this PR pretty minimal -- we can always add more logging down the line

File: packages/website/public/index.html [hunk 3]
```diff
@@ -1,53 +1,44 @@
...
-                js.src = "//connect.facebook.net/en_US/sdk.js#xfbml=1&version=v2.8&appId=1687545238205192";
-                fjs.parentNode.insertBefore(js, fjs);
-            }(document, 'script', 'facebook-jssdk'));
-        </script>
-        <div id="app"></div>
-        <!-- End Facebook SDK -->
-        <!-- Twitter SDK -->
-        <script>
-            window.twttr = (function(d, s, id) {
+<head>
```
Comment (LogvinovLeon) [inline packages/website/public/index.html]: Why did we left-shift that section?

File: packages/website/public/index.html [hunk 4]
```diff
@@ -56,23 +47,27 @@
...
+
+    <!-- Main -->
+    <script type="text/javascript" crossorigin="anonymous" src="/bundle.js" charset="utf-8"></script>
+</body>
 
-        <!-- Main -->
-        <script type="text/javascript" crossorigin="anonymous" src="/bundle.js" charset="utf-8"></script>
-    </body>
-</html>
+</html>
```
Comment (LogvinovLeon) [inline packages/website/public/index.html]: Why there is no newline now?

File: packages/website/ts/globals.d.ts [hunk 5]
```diff
@@ -11,6 +11,7 @@ declare module 'react-highlight';
 declare module 'react-recaptcha';
 declare module 'react-document-title';
 declare module 'ethereumjs-tx';
+declare module 'react-ga';
```
Comment (LogvinovLeon) [inline packages/website/ts/globals.d.ts]: 😢

Comment (tomhschmidt) [reply packages/website/ts/globals.d.ts]: lolz there's no typing afik

File: packages/website/public/index.html (hunk 3 above)
Comment (tomhschmidt) [reply packages/website/public/index.html]: Ran prettier again, and this is the same output

**Pull request merged**